packages/schema-registry-api/src/api.ts

```diff
@@ -32,15 +32,17 @@ const apiFetch = async <T>(req: string | Request, init: RequestInit = {}): Promi
       'Content-Type': 'application/vnd.schemaregistry.v1+json',
     },
   };
-  const res = await fetch(req, { ...defaultInit, ...init });
+
+  let res;
   let data;
   try {
+    res = await fetch(req, { ...defaultInit, ...init });
     data = await res.json();
   } catch (error) {
-    // Generic error to avoid including URL in message, as it may contain
-    // sensitive credentials.
-    throw TypeError('Schema registry responded with invalid JSON.');
+    // Use regex to sanitize error logging in order avoid basic auth credentials leakage
+    throw Error(error.toString().replace(/[^/]*@/g, '{CREDENTIALS}'));
```

Comment:
I believe some of the tests now need updating

Reply:
This [one](https://github.com/ovotech/bit-node-tools/blob/master/packages/schema-registry-api/test/integration.spec.ts#L116) at least

Reply:
Just changed it! Thanks! 🙂